### PR TITLE
fix/5501

### LINKED ISSUE #5501: Search highlight error due to error of  this.searchText ?
**Bootstraptable version(s) affected**:

1.18.1
**Description**  
Search highlight error due to **searchText**  option?

**Example** 
The online [official search highlight example](https://examples.bootstrap-table.com/#options/search-highlight.html) works as expected.  
The following example is constructed based on the official example: the bootstrap-table is initialized via JavaScript codes and the same set of data is retrieved through $.getJSON
https://live.bootstrap-table.com/code/hs749/5932
If opened in Firefox, Console panel of Developer Tools displays the following error msg:
> Uncaught TypeError: o.searchText is undefined
    value https://unpkg.com/bootstrap-table@1.18.1/dist/bootstrap-table.min.js:10
    value https://unpkg.com/bootstrap-table@1.18.1/dist/bootstrap-table.min.js:10
    value https://unpkg.com/bootstrap-table@1.18.1/dist/bootstrap-table.min.js:10
    value https://unpkg.com/bootstrap-table@1.18.1/dist/bootstrap-table.min.js:10
    bootstrapTable https://unpkg.com/bootstrap-table@1.18.1/dist/bootstrap-table.min.js:10
    jQuery 2
    bootstrapTable https://unpkg.com/bootstrap-table@1.18.1/dist/bootstrap-table.min.js:10
    <anonymous> https://live.bootstrap-table.com/example/options/search-highlight.html:32
    jQuery 8
    <anonymous> https://live.bootstrap-table.com/example/options/search-highlight.html:31

**Possible  (optional)**  
After replacing all the occurrences of **this.searchText** with **this.options.searchText** in line [1570](https://github.com/wenzhixin/bootstrap-table/blob/f638add58e5cf1f2c0251b2783aad861a6cc5f94/src/bootstrap-table.js#L1570) & 1571 in bootstrap-table.js (local copy), the error disappeared.

A quick search on **this.searchText** in [bootstrap-table.js](https://github.com/wenzhixin/bootstrap-table/blob/develop/src/bootstrap-table.js) returns 13 matches. Maybe a close examination on those 13 matches is warranted.

**Additional context**

### PR DESCRIPTION
**Pull request by UtechtDustin**
**Bug fix?**
yes

**New Feature?**
no

**Resolve an issue?**
Fix #5501

**Example(s)?**
Before: https://live.bootstrap-table.com/code/hs749/5957
After: https://live.bootstrap-table.com/code/UtechtDustin/5968

### CODE REVIEW

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -1567,7 +1567,7 @@ class BootstrapTable {
           this.options.undefinedText : value
       }
 
-      if (this.searchText !== '' && this.options.searchHighlight) {
+      if (this.searchText !== undefined && this.searchText !== '' && this.options.searchHighlight) {
```
Comment (djhvscf) [inline src/bootstrap-table.js:1570]: I think we could validate `searchText` as a `falsy` value then.

**Review by djhvscf — CHANGES_REQUESTED** [review]

File: src/bootstrap-table.js (hunk 1 above)
Comment (UtechtDustin) [reply src/bootstrap-table.js:1570]: I'm not sure what do you mean, could you please explain that further ?

Comment (hs749) [reply src/bootstrap-table.js:1570]: I agree on @UtechtDustin's fix. It's the simplest fix that addresses the **undefined** of **this.searchText**. When bootstrap-table is constructed, line [30](https://github.com/wenzhixin/bootstrap-table/blob/13ebc7e4a5bb95f8c6fbf0a1fc792ea71080f5b5/src/bootstrap-table.js#L30) is called, but it's the line 31 that initializes this.searchText.

Comment (djhvscf) [reply src/bootstrap-table.js:1570]: I mean if `searchText` can be `empty/null/undefined` we could validate it as `falsy` so:

`if (!this.searchText) {.....`

Comment (hs749) [reply src/bootstrap-table.js:1570]: Could this.searchText be set 0 if add **searchText** option in @UtechtDustin's example ?

$.getJSON('json/data1.json', function(tableData){
    $('#table').bootstrapTable({
   search: true,
   searchHighlight: true,
   searchText: 0,
   data: tableData
  });
 });

Comment (UtechtDustin) [reply src/bootstrap-table.js:1570]: @djhvscf @hs749 makes sense, i will change that later!

**Review by wenzhixin — APPROVED** [review]

**Pull request merged**